template/package.yaml

```diff
@@ -0,0 +1,111 @@
+# SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: Unlicense
+
+name:                templateDAO
+version:             0.0.0.0
+license:             MIT
+license-file:        LICENSE
+author:              Serokell, Tocqueville Group
+git:                 git@github.com:tqtezos/baseDAO.git
+copyright:           2020 Unlicense
+
+extra-source-files:
+- README.md
+
+description:         Please see the README on GitHub at <https://github.com/tqtezos/baseDAO/template#readme>
+
+verbatim:
+  cabal-version: 2.2
+
+default-extensions:
+  - AllowAmbiguousTypes
+  - BangPatterns
+  - BlockArguments
+  - ConstraintKinds
+  - DataKinds
+  - DefaultSignatures
+  - DeriveAnyClass
+  - DeriveDataTypeable
+  - DeriveFoldable
+  - DeriveFunctor
+  - DeriveGeneric
+  - DeriveTraversable
+  - DerivingStrategies
+  - DerivingVia
+  - EmptyCase
+  - ExistentialQuantification
+  - FlexibleContexts
+  - FlexibleInstances
+  - GADTs
+  - GeneralizedNewtypeDeriving
+  - LambdaCase
+  - MultiParamTypeClasses
+  - MultiWayIf
+  - NamedFieldPuns
+  - NegativeLiterals
+  - NoImplicitPrelude
+  - NumDecimals
+  - OverloadedLabels
+  - OverloadedStrings
+  - PatternSynonyms
+  - PolyKinds
+  - QuasiQuotes
+  - RankNTypes
+  - RecordWildCards
+  - RecursiveDo
+  - ScopedTypeVariables
+  - StandaloneDeriving
+  - StrictData
+  - TemplateHaskell
+  - TupleSections
+  - TypeApplications
+  - TypeFamilies
+  - TypeOperators
+  - UndecidableInstances
+  - UndecidableSuperClasses
+  - ViewPatterns
+
+ghc-options:
+  -Weverything
+  -Wno-missing-exported-signatures
+  -Wno-missing-import-lists
+  -Wno-missed-specialisations
+  -Wno-all-missed-specialisations
+  -Wno-unsafe
+  -Wno-safe
+  -Wno-missing-local-signatures
+  -Wno-monomorphism-restriction
+  -Wno-implicit-prelude
+
+dependencies:
+  - base >= 4.7 && < 5
+
+library:
```

Comment:
Is the library needed at all? I doubt anyone will ever use it as a library, so putting everything into executable seems simpler. AFAIU it's just one module.

Reply:
We can do this, though two things that worry me:
1. Someone can really create an intermediate library via defining polymorphic `ProposalMetadata` and `ContractExtra`. But such cases are probably rare and for advanced users, so we don't care.
2. Again, the whole `TemplateDAO` is a point of interest for the user, while `Main.hs` contains a lot of logic that may be considered complex and non-interesting, and mixing them will increase the cognitive load for the user (at least I feel so).
Maybe we could extract most of `Main.hs` to `CLI.hs` module and leave a three-liner at `Main.hs`, so that the user could concentrate on writing his logic? Of course, in a way that can be overridden if necessary. Then I think having one module is pretty fine.

Reply:
>  `TemplateDAO` is a point of interest for the user

If we see it this way, probably we can put everything that should be modified by a user into `TemplateDAO.hs` :shrug:
Or we can have `TemplateDAO.hs` and `CLI.hs` which should be modified.

Right now we have a place in `Main.hs` that needs to be updated, so places-to-be-updated are mixed anyway.

Reply:
This looks good, probably all we need to change in CLI is repo reference and contract name, so I'll just put them into `TemplateDAO.hs`.